common/libs/transaction.js

```diff
@@ -174,48 +168,44 @@ export async function formatTxInput(
     return {
       to,
       from: await wallet.getAddress(),
-      value: valueToHex(value),
+      value: valueToHex(new Ether(value)),
       data
     };
   } else {
     if (!token) {
       throw new Error('No matching token');
     }
     const bigAmount = new Big(value);
+    const ERC20Data = ERC20.transfer(to, bigAmount);
     return {
       to: token.address,
       from: await wallet.getAddress(),
       value: '0x0',
-      data: ERC20.transfer(to, toTokenUnit(bigAmount, token))
+      data: ERC20Data
     };
   }
 }
 
 export async function generateCompleteTransaction(
   wallet: BaseWallet,
   nodeLib: RPCNode,
-  gasPrice: string,
-  gasLimit: string,
+  gasPrice: Wei,
+  gasLimit: Big,
   chainId: number,
   transactionInput: TransactionInput
 ): Promise<CompleteTransaction> {
   const { token } = transactionInput;
-
   const formattedTx = await formatTxInput(wallet, transactionInput);
-
-  const from = await wallet.getAddress();
-
   const transaction: ExtendedRawTransaction = {
```

Comment:
Could destructure  `transaction` after line 199 to make the return object cleaner

Reply:
👍